server/sonar-server/src/main/java/org/sonar/server/projectlink/ws/DeleteAction.java

```diff
@@ -0,0 +1,103 @@
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.projectlink.ws;
+
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.api.web.UserRole;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.component.ComponentLinkDto;
+import org.sonar.server.user.UserSession;
+import org.sonar.server.ws.WsUtils;
+import org.sonarqube.ws.client.projectlinks.DeleteWsRequest;
+
+import static org.sonar.db.component.ComponentLinkDto.PROVIDED_TYPES;
+import static org.sonar.server.user.AbstractUserSession.insufficientPrivilegesException;
+import static org.sonarqube.ws.client.projectlinks.ProjectLinksWsParameters.ACTION_DELETE;
+import static org.sonarqube.ws.client.projectlinks.ProjectLinksWsParameters.PARAM_ID;
+
+public class DeleteAction implements ProjectLinksWsAction {
+  private final DbClient dbClient;
+  private final UserSession userSession;
+
+  public DeleteAction(DbClient dbClient, UserSession userSession) {
+    this.dbClient = dbClient;
+    this.userSession = userSession;
+  }
+
+  @Override
+  public void define(WebService.NewController context) {
+    WebService.NewAction action = context.createAction(ACTION_DELETE)
+      .setDescription("Delete existing project link.<br>" +
+        "Requires 'Administer' permission on the specified project, " +
+        "or global 'Administer' permission.")
+      .setHandler(this)
+      .setPost(true)
+      .setSince("6.1");
+
+    action.createParam(PARAM_ID)
+      .setRequired(true)
+      .setDescription("Link id")
+      .setExampleValue("17");
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    DeleteWsRequest deleteWsRequest = toDeleteWsRequest(request);
+    doHandle(deleteWsRequest);
+    response.noContent();
+  }
+
+  private DeleteWsRequest toDeleteWsRequest(Request request) {
+    return new DeleteWsRequest()
+      .setId(request.mandatoryParamAsLong(PARAM_ID));
+  }
+
+  private void doHandle(DeleteWsRequest request) {
+    DbSession dbSession = dbClient.openSession(false);
+    try {
+      long id = request.getId();
+      ComponentLinkDto link = dbClient.componentLinkDao().selectById(dbSession, id);
+
+      WsUtils.checkFound(link, "Link with id '%s' not found", id);
+      checkPermissions(link.getComponentUuid());
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'link' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)